python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -75,10 +75,12 @@ def write(
 
         def write_blocks_to_path():
             with self.open_output_stream(write_path) as file:
-                schema = BlockAccessor.for_block(blocks[0]).to_arrow().schema
+                tables = [BlockAccessor.for_block(block).to_arrow() for block in blocks]
+                schema = self._try_merge_nullable_fields(tables)
```

Comment:
Rather than introducing a new method, we could extend the existing `unify_schemas` function: https://github.com/ray-project/ray/blob/23cc23b7c295a2959682df785408e534095b2e19/python/ray/data/_internal/arrow_ops/transform_pyarrow.py#L54